src/search.js

```diff
@@ -0,0 +1,231 @@
+	"CREATE VIRTUAL TABLE IF NOT EXISTS texts USING fts4(content=\"logs\", nick, text)",
+	"CREATE TRIGGER IF NOT EXISTS logs_bu BEFORE UPDATE ON logs BEGIN DELETE FROM texts WHERE docid=old.rowid; END",
+	"CREATE TRIGGER IF NOT EXISTS logs_bu BEFORE UPDATE ON logs BEGIN DELETE FROM texts WHERE docid=old.rowid; END",
+	"CREATE TRIGGER IF NOT EXISTS logs_bd BEFORE DELETE ON logs BEGIN DELETE FROM texts WHERE docid=old.rowid; END",
+	"CREATE TRIGGER IF NOT EXISTS logs_ai AFTER INSERT ON logs BEGIN INSERT INTO texts(docid, nick, text) VALUES(new.rowid, new.nick, new.text); END"
+];
+
+var db = new Sqlite.Database(path.join(Helper.getLogsPath(), "search.db"));
+db.serialize(() => schema.forEach((line) => db.run(line)));
+
+var indexAll = () => {
+	db.serialize(() => {
+		const ClientManager = require("./clientManager");
+		var manager = new ClientManager();
+
+		var files = [];
+		manager.getUsers().forEach((user) => {
+			user = manager.readUserConfig(user);
+			user.networks.forEach(function(network) {
+				var root = Helper.getUserLogsPath(user.user, network.host);
+				fs.readdirSync(root).forEach((filename) => {
+					if (path.extname(filename) === ".log") {
+						var chan = path.basename(filename, ".log");
+						if (Helper.config.search.indexLobby === true || chan !== network.host) {
+							files.push({
+								user: user.user,
+								network: network.host,
+								chan: chan,
+								path: path.join(root, filename)
+							});
+						}
+					}
+				});
+			});
+		});
+		Promise.each(files, indexFile);
+	});
+};
+
+var indexFile = (file) => new Promise((resolve, reject) => {
+	log.info("Indexing", file.path);
+	fs.createReadStream(file.path, {flags: "r"})
+		.pipe(es.split())
+		.pipe(es.map((line, cb) => {
+			if (line) {
+				var endOfTime = line.indexOf("]");
+				var time = moment(line.substr(1, endOfTime - 1), format);
+				if (line[endOfTime + 2] === "<") {
+					var endOfNick = line.indexOf(">", endOfTime + 2);
+					db.run("INSERT INTO logs(user, network, chan, time, nick, text) VALUES(?, ?, ?, ?, ?, ?)",
+						file.user, file.network, file.chan, time.unix(),
+						line.substr(endOfTime + 3, endOfNick - endOfTime - 3),
+						line.substr(endOfNick + 2));
+				} else if (Helper.config.search.indexOnlyChanMessage === false || (Helper.config.search.indexLobby === true && file.chan === file.network)) {
+					db.run("INSERT INTO logs(user, network, chan, time, nick, text) VALUES(?, ?, ?, ?, ?, ?)",
+						file.user, file.network, file.chan, time.unix(),
+						"",
+						line.substr(endOfTime + 2));
+				}
+			}
+			cb(null, line);
+		}))
+		.on("error", (err) => reject(err))
+		.on("end", () => resolve());
+});
+
+var index = (user, network, chan, time, nick, text) => {
+	console.log(user, network, chan, time, nick, text);
+	return Promise.resolve(db)
+		.then(() => Sqlite.run("INSERT INTO logs(user, network, chan, time, nick, text) VALUES(?, ?, ?, ?, ?, ?)",
```

Comment:
Is the code correct here, should it actually be `Sqlite.run`?

Reply:
A commit is missing, sorry.